### ISSUE TITLE
[lxqt-panel] NET::WMIconGeometry is causing high CPU and memory usage

### ISSUE THREAD

**Issue by PCMan** [lxde/lxqt#587]
In LxQtTaskGroup::onWindowChanged(() defined in lxqttaskgroup.cpp, I saw the following FIXME.
...
        // FIXME: NET::WMIconGeometry is causing high CPU and memory usage
        if (prop.testFlag(NET::WMIcon) /*|| prop.testFlag(NET::WMIconGeometry)*/)
            button->updateIcon();

I guess the problem is caused by LxQtTaskButton::refreshIconGeometry().
When the panel is moved or resized, we tried to set icon geometry for buttons.
This triggers LxQtTaskGroup::onWindowChanged().
In this case, prop.testFlag(NET::WMIconGeometry) returns true, so button->updateIcon() will be called.
This might trigger the resize of task button, and its parent widget, the panel.
If unfortunately the size of the panel is changed, refreshIconGeometry() will be called again.
This recursion will only stop when the size of icons and the panel no longer change.
I don't have the time for further debugging, but based on code review, I believe that this might be the potential cause of high CPU usage.

**Comment by palinek** [comment]
It could be that I'm missing something, but why do we set the window icon geometry at all?
After commentig the whole `LxQtTaskButton::refreshIconGeometry` I don't see any difference in behaviour/visual of taskbar:
```diff
--- a/plugin-taskbar/lxqttaskbutton.cpp
+++ b/plugin-taskbar/lxqttaskbutton.cpp
@@ -134,6 +134,7 @@ void LxQtTaskButton::updateIcon()
  ************************************************/
 void LxQtTaskButton::refreshIconGeometry()
 {
+    /*
     QRect rect = geometry();
     QPoint globalPos = parentTaskBar()->mapToGlobal(pos());
     rect.moveTo(globalPos);
@@ -149,6 +150,7 @@ void LxQtTaskButton::refreshIconGeometry()
     nrect.size.height = rect.height();
     nrect.size.width = rect.width();
     info.setIconGeometry(nrect);
+    */
 }
 
 /************************************************
```

**Comment by PCMan** [comment]
The icon geometry stuff is for WMs which support animations when minimizing windows.
When you minimize a window, many WMs use that rectangle as the destination of their animations.
For example, openbox is one of these WMs.

**Comment by palinek** [comment]
I've realized that already...(sorry for stupid questions :) )

**Comment by palinek** [comment]
..while debuging i was realizing a smash of `NET::WMIconGeometry` and still didn't get where and why they are commig => the problem here is also if we have more than one plugin-taskbar active, each instance is setting the icon geometry and this leads to signal flood.

**Comment by palinek** [comment]
I believe this has been solved in lxde/lxqt-panel#166 ...
